### ISSUE TITLE
some questions about the test accuracy

### ISSUE THREAD

**Issue by WwwwJX** [voxelmorph/voxelmorph#373]
Sorry to trouble you.
I meet a question about the test accuracy. According your code, I should write a pair.txt, and the pair.txt should be : test_moving.nii path  train_fixed.nii path.
But the test accuray of it is 66.
However when I make the pair.txt like :    train_fixed.nii path  test_moving.nii   the accuracy is 69, it's higher than the first. But I tnink it's unreasonable. Can you tell me why?

**Comment by tphankr** [comment]
1) My opinion: @WwwwJX . It is easyly understand becuase the second time Image registration output is similar than with Groudn Truth. The problem: we have to determine exactly where the image is the "train_fixed.nii path", where is the "test_moving.nii".


2)  @WwwwJX, Can you capture your pair.txt file or comment when you run your test code. because I am confused to run this test step?. I mean : How do you setting with command line args in the test.py file. or **How does look like the pair.txt file for the "test_moving.nii" and "train_fixed.nii" path?** Aslo  --img-suffix, --seg-suffix, --img-prefix, --seg-predfix that is arguments in the test.py file. Thanks.

- ------------------------------------

3) Dear @adalca, @ahoopes and everyone,
For example, with the brain data from( https://github.com/adalca/medical-datasets/blob/master/neurite-oasis.md)
Please guide us to something, thank you @adalca @ahoopes .

+ About setting --pairs, --img-suffix, ...in the test.py file. I also tried to do **but I do not image yet about how do we create the --pairs ( pairs.txt)** to have the pair image input although read your comments the "--pairs" is the pairs.txt file of pair image input include fixed image and moving image? Does it mean a .txt file, like this:
![image](https://user-images.githubusercontent.com/37217354/147455448-667962c5-33f6-4b2c-9294-e69143c78a93.png)

My status: 
- ![image](https://user-images.githubusercontent.com/37217354/147456665-d2e86af0-19be-4403-8aa7-5b97a8b5c08c.png)



Thank you @adalca @ahoopes and @WwwwJX and everyone.

**Comment by WwwwJX** [comment]
The input pairs have at least two images. Source image and Target image, so you should make your input.txt like 
source target1image
source target2image
... and so on
But it's my opinion about the project and maybe it's not right. As your answer about my question, I still don't know why. Please tell me the reason for this result.
@adalca @ahoopes @tphankr

**Comment by ahoopes** [comment]
Right. Each line in the 'pairs' text file should define a (moving, fixed) image pair to align, for example:

```
OASIS_OAS1_0001_MR1 OASIS_OAS1_0002_MR1
OASIS_OAS1_0003_MR1 OASIS_OAS1_0004_MR1
...
```

Then you could, for example, use the corresponding 35-label segmentations to compute accuracy using a command similar to:

```
python test.py  \
    --pairs pairs.txt  \
    --img-prefix /path/to/oasis/data/  \
    --seg-prefix /path/to/oasis/data/  \
    --img-suffix /aligned_norm.nii.gz  \
    --seg-suffix /aligned_seg35.nii.gz  \
    ...
```

@WwwwJX are you asking why the Dice scores differ depending on the deformation direction? or why these scores are in the 60s?

The scores will differ depending on the direction for multiple reasons. The two *individually-computed* warps are not perfect inverses of each other since the moving image differs in both cases. Even if the two deformations were perfect inverses, the moving segmentation is still different and therefore resampling noise will be introduced.

If you're worried about the Dice scores, you could modify `test.py` to save the moved images/segmentations and look into which label(s) are poorly aligned. Maybe it's an issue with the model or maybe not. If the alignment result is bad, we can troubleshoot further, but keep in mind Dice is not a perfect measure of registration accuracy. It's possible the alignments are quite good in reality, so you should look for yourself.

**Comment by tphankr** [comment]
Thank you for your reply @ahoopes.
These day, I also try to think about the @WwwwJX's  problem but not clearly answer to @WwwwJX. Thank you @ahoopes.

@ahoopes I will do the step immediately and report again if have problem.

Thanks all.

**Comment by WwwwJX** [comment]
@ahoopes 
Thanks for your answer sincerely. 
And I have a question about the input.txt. According your paper, you let one image as fixed image and let the others as moving inage. So I think the input.txt should:
moving_1, fixed
moving_2, fixed
moving_3, fixed
...
why you let the txt as 
OASIS_OAS1_0001_MR1 OASIS_OAS1_0002_MR1
OASIS_OAS1_0003_MR1 OASIS_OAS1_0004_MR1

I hope you can tell me why

**Comment by ahoopes** [comment]
@WwwwJX which paper are you referring to? While you might have only one fixed image for atlas-based (subject-to-atlas) registration models, you can use any arbitrary image pair at test-time for subject-to-subject registration models.
